src/web/app/components/course-edit-form/course-edit-form.component.spec.ts

```diff
@@ -215,4 +216,188 @@ describe('CourseEditFormComponent', () => {
     expect(mockModalRef.componentInstance.activeCourses[0]).toEqual(testCourse);
     expect(mockModalRef.componentInstance.courseToFeedbackSession[testCourse.courseId]).toEqual([testFeedbackSession]);
   });
+
+  it('should handle errors in copyCourseHandler when promise is rejected', () => {
+    component.formModel = DEFAULT_COURSE_ADD_FORM_MODEL();
+    component.formMode = CourseEditFormMode.ADD;
+    fixture.detectChanges();
+
+    const mockModalRef: any = createMockNgbModalRef({
+      isCopyFromOtherSession: false,
+      courses: [],
+      courseToFeedbackSession: {},
+      fetchFeedbackSessionsEvent: new EventEmitter<string>(),
+    }, Promise.reject(new Error(errorMssg)));
+    jest.spyOn(ngbModal, 'open').mockReturnValue(mockModalRef);
+
+    component.copyCourseHandler();
+
+    expect(spyStatusMessageService.showErrorToast()).toBe(errorMssg);
+  });
+
+  it('should handle errors in copyCourseHandler when observable throws an error', () => {
+    component.formModel = DEFAULT_COURSE_ADD_FORM_MODEL();
+    component.formMode = CourseEditFormMode.ADD;
+    fixture.detectChanges();
+
+    const mockModalRef: any = createMockNgbModalRef({
+      isCopyFromOtherSession: false,
+      courses: [],
+      courseToFeedbackSession: {},
+      fetchFeedbackSessionsEvent: new Observable<number>((observer) => {
+        observer.error({
+          error: { message: errorMssg },
+        });
+      }),
+    });
+    jest.spyOn(ngbModal, 'open').mockReturnValue(mockModalRef);
+
+    component.copyCourseHandler();
+
+    expect(spyStatusMessageService.showErrorToast()).toBe(errorMssg);
+  });
+
+  it('should set isEditing to true when editModel exists', () => {
+    component.editModel = DEFAULT_COURSE_EDIT_FORM_MODEL();
+
+    component.setIsEditing(true);
+
+    expect(component.editModel?.isEditing).toBe(true);
+  });
+
+  it('should set isEditing to false when editModel exists', () => {
+    component.editModel = DEFAULT_COURSE_EDIT_FORM_MODEL();
+
+    component.setIsEditing(false);
+
+    expect(component.editModel?.isEditing).toBe(false);
+  });
+
+  it('should not modify isEditing when editModel is undefined', () => {
+    component.editModel = undefined;
+
+    component.setIsEditing(true);
+
+    expect(component.editModel).toBeUndefined();
```

Comment:
this is also always true since we set it as undefined, we should be setting isEditing to false, and then checking after setIsEditing if it remains true